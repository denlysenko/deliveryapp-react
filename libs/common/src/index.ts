export * from './lib/constants';
export * from './lib/errors';
export * from './lib/roles';
export * from './lib/app-menu';
export * from './lib/messages';
export * from './lib/payment-methods';
