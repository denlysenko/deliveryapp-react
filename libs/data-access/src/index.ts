import * as authClient from './lib/api/auth';
import * as usersClient from './lib/api/users';
import * as ordersClient from './lib/api/orders';
import * as paymentsClient from './lib/api/payments';

export { authClient, usersClient, ordersClient, paymentsClient };

export * from './lib/models/auth';
export * from './lib/models/user';
export * from './lib/models/validation-error';
export * from './lib/models/order';
export * from './lib/models/list-response';
export * from './lib/models/orders-filter';
export * from './lib/models/payment';
export * from './lib/models/payments-filter';
export * from './lib/models/users-filter';

export * from './lib/context/auth';
export * from './lib/context/orders';
export * from './lib/context/payments';
