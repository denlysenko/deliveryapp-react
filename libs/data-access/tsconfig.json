{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "jsx": "react",
    "esModuleInterop": true,
    "types": ["node", "jest"]
  },
  "include": ["**/*.ts"]
}
