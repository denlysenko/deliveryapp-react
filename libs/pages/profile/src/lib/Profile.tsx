/* eslint-disable @typescript-eslint/no-non-null-assertion */
import React, { useState, useRef } from 'react';

import { useFormik } from 'formik';
import { Button } from 'primereact/button';
import { Growl } from 'primereact/growl';

import * as Yup from 'yup';
import { isNil, has, omit } from 'lodash-es';

import { Roles, ERRORS, MESSAGES } from '@deliveryapp/common';
import {
  useAuth,
  UserDTO,
  Address,
  BankDetails,
  usersClient,
  AuthActionTypes
} from '@deliveryapp/data-access';
import { handleValidationError } from '@deliveryapp/utils';

import { StyledProfile } from './StyledProfile';
import { ContactsForm } from './ContactsForm/ContactsForm';
import { PasswordForm } from './PasswordForm/PasswordForm';
import { AddressForm } from './AddressForm/AddressForm';
import { BankDetailsForm } from './BankDetailsForm/BankDetailsForm';

export interface ProfileFormValues extends Omit<UserDTO, 'role' | 'password'> {
  address?: Address;
  bankDetails?: BankDetails;
}

const ValidationSchema = Yup.object().shape({
  email: Yup.string()
    .required(ERRORS.REQUIRED_FIELD)
    .email(ERRORS.INVALID_EMAIL)
});

export const Profile = () => {
  const growl = useRef<Growl>(null);
  const [loading, setLoading] = useState(false);
  const [{ user }, dispatch] = useAuth();

  const formik = useFormik<ProfileFormValues>({
    initialValues: {
      firstName: !isNil(user!.firstName) ? user!.firstName : '',
      lastName: !isNil(user!.lastName) ? user!.lastName : '',
      email: user!.email || '',
      company: !isNil(user!.company) ? user!.company : '',
      phone: !isNil(user!.phone) ? user!.phone : '',
      ...(!isNil(user) &&
        user.role === Roles.CLIENT && {
          address: {
            country: !isNil(user.address) ? user.address.country : '',
            city: !isNil(user.address) ? user.address.city : '',
            street: !isNil(user.address) ? user.address.street : '',
            house: !isNil(user.address) ? user.address.house : ''
          },
          bankDetails: {
            name: !isNil(user.bankDetails) ? user.bankDetails.name : '',
            accountNumber: !isNil(user.bankDetails)
              ? user.bankDetails.accountNumber
              : '',
            bin: !isNil(user.bankDetails) ? user.bankDetails.bin : '',
            swift: !isNil(user.bankDetails) ? user.bankDetails.swift : ''
          }
        })
    },
    initialStatus: {
      apiErrors: {}
    },
    validationSchema: ValidationSchema,
    onSubmit: async (values) => {
      setLoading(true);

      try {
        await usersClient.updateProfile(values);
        dispatch({
          type: AuthActionTypes.USER_LOADED,
          payload: {
            ...user!,
            ...values,
            ...(user!.role === Roles.CLIENT && {
              address: {
                ...user!.address,
                ...values.address!
              },
              bankDetails: {
                ...user!.bankDetails,
                ...values.bankDetails!
              }
            })
          }
        });
        setLoading(false);
        !isNil(growl.current) &&
          growl.current.show({
            severity: 'success',
            summary: MESSAGES.UPDATE_PROFILE_SUCCESS,
            closable: false
          });
      } catch (error) {
        setLoading(false);
        handleValidationError<ProfileFormValues>(error.response.data, formik);
      }
    }
  });

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const name = event.target.name;

    if (has(formik.status.apiErrors, name)) {
      formik.setStatus({
        apiErrors: omit(formik.status.apiErrors, name)
      });
    }

    formik.handleChange(event);
  };

  return (
    <StyledProfile>
      <Growl ref={growl} />
      <div className="p-grid">
        <div
          className={`p-col-12 ${user?.role !== Roles.CLIENT ? 'p-lg-4' : ''}`}
        >
          <div className="card no-margin-bottom">
            <h2>Profile</h2>
          </div>
        </div>
        <form
          id="profileForm"
          data-testid="profile-form"
          onSubmit={formik.handleSubmit}
        >
          <div className="p-col-12 p-lg-4">
            <div className="card">
              <ContactsForm
                values={formik.values}
                errors={formik.errors}
                apiErrors={formik.status.apiErrors}
                touched={formik.touched}
                handleChange={handleChange}
              />
              <PasswordForm />
            </div>
          </div>
          {user?.role === Roles.CLIENT && (
            <div className="p-col-12 p-lg-4">
              <div className="card">
                <AddressForm
                  address={formik.values.address}
                  handleChange={formik.handleChange}
                />
              </div>
            </div>
          )}
          {user?.role === Roles.CLIENT && (
            <div className="p-col-12 p-lg-4">
              <div className="card">
                <BankDetailsForm
                  bankDetails={formik.values.bankDetails}
                  handleChange={formik.handleChange}
                />
              </div>
            </div>
          )}
        </form>
        <div
          className={`p-col-12 ${user?.role !== Roles.CLIENT ? 'p-lg-4' : ''}`}
        >
          <div className="card no-margin-bottom center">
            <Button
              label="Save"
              type="submit"
              form="profileForm"
              className="p-button-raised"
              data-testid="save-profile"
              disabled={loading}
            />
          </div>
        </div>
      </div>
    </StyledProfile>
  );
};
