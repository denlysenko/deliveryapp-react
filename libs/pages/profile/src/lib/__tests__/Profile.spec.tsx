import React from 'react';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';

import { ERRORS, Roles } from '@deliveryapp/common';
import { AuthProvider, usersClient, useAuth } from '@deliveryapp/data-access';
import { user } from '@deliveryapp/testing';

import { Profile } from '../Profile';

jest.mock('@deliveryapp/data-access', () => ({
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ...jest.requireActual<any>('@deliveryapp/data-access'),
  useAuth: jest.fn().mockImplementation(() => [{ user }, jest.fn()])
}));

describe('Profile page', () => {
  beforeEach(() => {
    jest
      .spyOn(usersClient, 'updateProfile')
      .mockImplementation(() => Promise.resolve({ data: user }));
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render successfully', () => {
    const { baseElement } = render(
      <AuthProvider>
        <Profile />
      </AuthProvider>
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should not render address & bankDetails', () => {
    (useAuth as jest.MockedFunction<
      typeof useAuth
    >).mockImplementationOnce(() => [
      { user: { ...user, role: Roles.ADMIN }, isLoggedIn: true },
      jest.fn()
    ]);

    const { baseElement } = render(
      <AuthProvider>
        <Profile />
      </AuthProvider>
    );
    expect(baseElement).toMatchSnapshot();
  });

  describe('Validations', () => {
    describe('email', () => {
      it('should display required error', async () => {
        const { container } = render(
          <AuthProvider>
            <Profile />
          </AuthProvider>
        );

        fireEvent.change(screen.getByTestId('email'), {
          target: {
            value: ''
          }
        });

        fireEvent.submit(screen.getByTestId('profile-form'));

        await waitFor(() => {
          expect(container.querySelector('#email-error')).toHaveTextContent(
            ERRORS.REQUIRED_FIELD
          );
        });
      });

      it('should display email error', async () => {
        const { container } = render(
          <AuthProvider>
            <Profile />
          </AuthProvider>
        );

        fireEvent.change(screen.getByTestId('email'), {
          target: {
            value: 'test'
          }
        });

        fireEvent.submit(screen.getByTestId('profile-form'));

        await waitFor(() => {
          expect(container.querySelector('#email-error')).toHaveTextContent(
            ERRORS.INVALID_EMAIL
          );
        });
      });

      it('should not display any errors', async () => {
        const { container } = render(
          <AuthProvider>
            <Profile />
          </AuthProvider>
        );

        fireEvent.submit(screen.getByTestId('profile-form'));

        await waitFor(() => {
          expect(
            container.querySelector('#email-error')
          ).not.toBeInTheDocument();
        });
      });
    });
  });

  describe('Submitting', () => {
    it('should be successful', async () => {
      const firstName = 'John';
      const lastName = 'Doe';
      const country = 'Ukraine';
      const name = 'The Bank';

      render(
        <AuthProvider>
          <Profile />
        </AuthProvider>
      );

      fireEvent.change(screen.getByTestId('firstName'), {
        target: {
          value: 'John'
        }
      });

      fireEvent.change(screen.getByTestId('lastName'), {
        target: {
          value: 'Doe'
        }
      });

      fireEvent.change(screen.getByTestId('country'), {
        target: {
          value: 'Ukraine'
        }
      });

      fireEvent.change(screen.getByTestId('name'), {
        target: {
          value: 'The Bank'
        }
      });

      fireEvent.submit(screen.getByTestId('profile-form'));

      const { id, role, createdAt, ...rest } = user;

      await waitFor(() => {
        expect(usersClient.updateProfile).toBeCalledTimes(1);
      });

      expect(usersClient.updateProfile).toBeCalledWith({
        ...rest,
        firstName,
        lastName,
        address: {
          city: '',
          country,
          house: '',
          street: ''
        },
        bankDetails: {
          accountNumber: '',
          bin: '',
          name,
          swift: ''
        }
      });
    });

    it('should have API error', async () => {
      const error = {
        errors: [{ path: 'email', message: ['Error'] }]
      };

      jest.spyOn(usersClient, 'updateProfile').mockImplementationOnce(() =>
        Promise.reject({
          response: {
            data: error
          }
        })
      );

      render(
        <AuthProvider>
          <Profile />
        </AuthProvider>
      );

      fireEvent.submit(screen.getByTestId('profile-form'));

      expect(
        await screen.findByText(error.errors[0].message[0])
      ).toBeInTheDocument();

      fireEvent.change(screen.getByTestId('email'), {
        target: {
          value: 'fixed@test.com'
        }
      });

      await waitFor(() => {
        expect(
          screen.queryByText(error.errors[0].message[0])
        ).not.toBeInTheDocument();
      });
    });
  });
});
