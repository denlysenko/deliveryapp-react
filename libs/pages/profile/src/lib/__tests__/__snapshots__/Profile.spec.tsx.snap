// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page should not render address & bankDetails 1`] = `
<body>
  <div>
    <div
      class="sc-AxjAm sc-AxirZ fwLWdz"
    >
      <div
        class="p-growl p-component p-growl-topright"
      >
        <div />
      </div>
      <div
        class="p-grid"
      >
        <div
          class="p-col-12 p-lg-4"
        >
          <div
            class="card no-margin-bottom"
          >
            <h2>
              Profile
            </h2>
          </div>
        </div>
        <form
          data-testid="profile-form"
          id="profileForm"
        >
          <div
            class="p-col-12 p-lg-4"
          >
            <div
              class="card"
            >
              <div
                class="sc-AxjAm icXODU"
              >
                <h3>
                  Contacts
                </h3>
                <div
                  class="p-grid"
                >
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component p-filled"
                        data-testid="firstName"
                        id="firstName"
                        name="firstName"
                        value="User"
                      />
                      <label
                        for="firstName"
                      >
                        First Name
                      </label>
                      <i
                        class="fa fa-user-circle-o"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component p-filled"
                        data-testid="lastName"
                        id="lastName"
                        name="lastName"
                        value="Test"
                      />
                      <label
                        for="lastName"
                      >
                        Last Name
                      </label>
                      <i
                        class="fa fa-user-circle-o"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component p-filled"
                        data-testid="company"
                        id="company"
                        name="company"
                        value="company"
                      />
                      <label
                        for="company"
                      >
                        Company
                      </label>
                      <i
                        class="fa fa-building-o"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component p-filled"
                        data-testid="email"
                        id="email"
                        name="email"
                        value="test@test.com"
                      />
                      <label
                        for="email"
                      >
                        Email
                      </label>
                      <i
                        class="fa fa-envelope-o"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component"
                        id="phone"
                        name="phone"
                        type="text"
                      />
                      <label
                        for="phone"
                      >
                        Phone
                      </label>
                      <i
                        class="fa fa-phone"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="sc-AxjAm icXODU"
              >
                <h3>
                  Change Password
                </h3>
                <div
                  class="p-grid"
                >
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component"
                        data-testid="oldPassword"
                        id="oldPassword"
                        name="oldPassword"
                        type="password"
                        value=""
                      />
                      <label
                        for="oldPassword"
                      >
                        Current Password
                      </label>
                      <i
                        class="fa fa-lock"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component"
                        data-testid="newPassword"
                        id="newPassword"
                        name="newPassword"
                        type="password"
                        value=""
                      />
                      <label>
                        New Password
                      </label>
                      <i
                        class="fa fa-lock"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component"
                        data-testid="confirmPassword"
                        id="confirmPassword"
                        name="confirmPassword"
                        type="password"
                        value=""
                      />
                      <label
                        for="confirmPassword"
                      >
                        Confirm Password
                      </label>
                      <i
                        class="fa fa-lock"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row center"
                  >
                    <button
                      class="p-button p-component p-button-raised p-button-text-only"
                      data-testid="change-password"
                      type="button"
                    >
                      <span
                        class="p-button-text p-c"
                      >
                        Change Password
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
        <div
          class="p-col-12 p-lg-4"
        >
          <div
            class="card no-margin-bottom center"
          >
            <button
              class="p-button p-component p-button-raised p-button-text-only"
              data-testid="save-profile"
              form="profileForm"
              type="submit"
            >
              <span
                class="p-button-text p-c"
              >
                Save
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Profile page should render successfully 1`] = `
<body>
  <div>
    <div
      class="sc-AxjAm sc-AxirZ fwLWdz"
    >
      <div
        class="p-growl p-component p-growl-topright"
      >
        <div />
      </div>
      <div
        class="p-grid"
      >
        <div
          class="p-col-12 "
        >
          <div
            class="card no-margin-bottom"
          >
            <h2>
              Profile
            </h2>
          </div>
        </div>
        <form
          data-testid="profile-form"
          id="profileForm"
        >
          <div
            class="p-col-12 p-lg-4"
          >
            <div
              class="card"
            >
              <div
                class="sc-AxjAm icXODU"
              >
                <h3>
                  Contacts
                </h3>
                <div
                  class="p-grid"
                >
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component p-filled"
                        data-testid="firstName"
                        id="firstName"
                        name="firstName"
                        value="User"
                      />
                      <label
                        for="firstName"
                      >
                        First Name
                      </label>
                      <i
                        class="fa fa-user-circle-o"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component p-filled"
                        data-testid="lastName"
                        id="lastName"
                        name="lastName"
                        value="Test"
                      />
                      <label
                        for="lastName"
                      >
                        Last Name
                      </label>
                      <i
                        class="fa fa-user-circle-o"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component p-filled"
                        data-testid="company"
                        id="company"
                        name="company"
                        value="company"
                      />
                      <label
                        for="company"
                      >
                        Company
                      </label>
                      <i
                        class="fa fa-building-o"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component p-filled"
                        data-testid="email"
                        id="email"
                        name="email"
                        value="test@test.com"
                      />
                      <label
                        for="email"
                      >
                        Email
                      </label>
                      <i
                        class="fa fa-envelope-o"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component"
                        id="phone"
                        name="phone"
                        type="text"
                      />
                      <label
                        for="phone"
                      >
                        Phone
                      </label>
                      <i
                        class="fa fa-phone"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="sc-AxjAm icXODU"
              >
                <h3>
                  Change Password
                </h3>
                <div
                  class="p-grid"
                >
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component"
                        data-testid="oldPassword"
                        id="oldPassword"
                        name="oldPassword"
                        type="password"
                        value=""
                      />
                      <label
                        for="oldPassword"
                      >
                        Current Password
                      </label>
                      <i
                        class="fa fa-lock"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component"
                        data-testid="newPassword"
                        id="newPassword"
                        name="newPassword"
                        type="password"
                        value=""
                      />
                      <label>
                        New Password
                      </label>
                      <i
                        class="fa fa-lock"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component"
                        data-testid="confirmPassword"
                        id="confirmPassword"
                        name="confirmPassword"
                        type="password"
                        value=""
                      />
                      <label
                        for="confirmPassword"
                      >
                        Confirm Password
                      </label>
                      <i
                        class="fa fa-lock"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row center"
                  >
                    <button
                      class="p-button p-component p-button-raised p-button-text-only"
                      data-testid="change-password"
                      type="button"
                    >
                      <span
                        class="p-button-text p-c"
                      >
                        Change Password
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="p-col-12 p-lg-4"
          >
            <div
              class="card"
            >
              <div
                class="sc-AxjAm icXODU"
              >
                <h3>
                  Address
                </h3>
                <div
                  class="p-grid"
                >
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component"
                        data-testid="country"
                        id="country"
                        name="address.country"
                        type="text"
                        value=""
                      />
                      <label
                        for="country"
                      >
                        Country
                      </label>
                      <i
                        class="fa fa-globe"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component"
                        data-testid="city"
                        id="city"
                        name="address.city"
                        type="text"
                        value=""
                      />
                      <label
                        for="city"
                      >
                        City
                      </label>
                      <i
                        class="fa fa-globe"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component"
                        data-testid="street"
                        id="street"
                        name="address.street"
                        type="text"
                        value=""
                      />
                      <label
                        for="street"
                      >
                        Street
                      </label>
                      <i
                        class="fa fa-building-o"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component"
                        data-testid="house"
                        id="house"
                        name="address.house"
                        type="text"
                        value=""
                      />
                      <label
                        for="house"
                      >
                        House
                      </label>
                      <i
                        class="fa fa-building-o"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="p-col-12 p-lg-4"
          >
            <div
              class="card"
            >
              <div
                class="sc-AxjAm icXODU"
              >
                <h3>
                  Bank Details
                </h3>
                <div
                  class="p-grid"
                >
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component"
                        data-testid="name"
                        id="name"
                        name="bankDetails.name"
                        type="text"
                        value=""
                      />
                      <label
                        for="name"
                      >
                        Name
                      </label>
                      <i
                        class="fa fa-building-o"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component"
                        id="accountNumber"
                        name="bankDetails.accountNumber"
                        type="text"
                      />
                      <label
                        for="accountNumber"
                      >
                        Account Number
                      </label>
                      <i
                        class="fa fa-credit-card"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component"
                        id="bin"
                        name="bankDetails.bin"
                        type="text"
                      />
                      <label
                        for="bin"
                      >
                        BIN
                      </label>
                      <i
                        class="fa fa-credit-card"
                      />
                    </div>
                  </div>
                  <div
                    class="p-col-12 row"
                  >
                    <div
                      class="input-wrapper p-float-label"
                    >
                      <input
                        class="p-inputtext p-component"
                        id="swift"
                        name="bankDetails.swift"
                        type="text"
                      />
                      <label
                        for="swift"
                      >
                        SWIFT
                      </label>
                      <i
                        class="fa fa-credit-card"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
        <div
          class="p-col-12 "
        >
          <div
            class="card no-margin-bottom center"
          >
            <button
              class="p-button p-component p-button-raised p-button-text-only"
              data-testid="save-profile"
              form="profileForm"
              type="submit"
            >
              <span
                class="p-button-text p-c"
              >
                Save
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
